src/ActionServer_FJT.c

```diff
@@ -495,18 +495,59 @@ void Ros_ActionServer_FJT_Goal_Complete(GOAL_END_TYPE goal_end_type)
-            if (posTolerance == 0.0) //user did NOT provide a tolerance
-                posTolerance = DEFAULT_FJT_GOAL_POSITION_TOLERANCE;
+            for (size_t selected_position_idx = 0; selected_position_idx < feedback_FollowJointTrajectory.feedback.joint_names.size; selected_position_idx += 1)
+            {
+                rosidl_runtime_c__String selected_position_name = feedback_FollowJointTrajectory.feedback.joint_names.data[selected_position_idx];
+                if (rosidl_runtime_c__String__are_equal(&selected_position_name,&selected_tolerance_name))
+                {
+                    Ros_Debug_BroadcastMsg("FJT using position tolerance: %s", selected_position_name);
+                    posTolerance[selected_position_idx] = g_actionServer_FJT_SendGoal_Request.goal.goal_tolerance.data[selected_tolerance_idx].position;
+                    break;
+                }
+            }
+        }
 
-            if (diff > posTolerance)
+        double finalPositions[maxAxes];
+        bzero(finalPositions, sizeof(finalPositions));
+        size_t trajectorySizeOffset = g_actionServer_FJT_SendGoal_Request.goal.trajectory.points.size - 1;
+        for (size_t selected_final_position_idx = 0; selected_final_position_idx < g_actionServer_FJT_SendGoal_Request.goal.trajectory.joint_names.size; selected_final_position_idx += 1)
+        {
+            rosidl_runtime_c__String selected_final_position_name = g_actionServer_FJT_SendGoal_Request.goal.trajectory.joint_names.data[selected_final_position_idx];
+            for (size_t selected_position_idx = 0; selected_position_idx < feedback_FollowJointTrajectory.feedback.joint_names.size; selected_position_idx += 1)
+            {
+                rosidl_runtime_c__String selected_position_name = feedback_FollowJointTrajectory.feedback.joint_names.data[selected_position_idx];
+                if (rosidl_runtime_c__String__are_equal(&selected_position_name, &selected_final_position_name))
+                {
+                    finalPositions[selected_position_idx] = g_actionServer_FJT_SendGoal_Request.goal.trajectory.points.data[trajectorySizeOffset].positions.data[selected_final_position_idx];
+                    Ros_Debug_BroadcastMsg("Final Position Goal: %s, %f", selected_position_name, finalPositions[selected_position_idx]);
+                    break;
+                }
+            }
+        }
+
+        for (int axis = 0; axis < maxAxes; axis += 1)
+        {
+            diff = fabs(finalPositions[axis] - feedback_FollowJointTrajectory.feedback.actual.positions.data[axis]);
+            // Ros_Debug_BroadcastMsg("desired - actual: %d, %d", (int)(1000 * finalPositions[axis]), (int)(1000 * feedback_FollowJointTrajectory.feedback.actual.positions.data[axis]));
+            double chosen_posTolerance = posTolerance[axis];
+            if (chosen_posTolerance == 0.0) //user did NOT provide a tolerance
+                chosen_posTolerance = DEFAULT_FJT_GOAL_POSITION_TOLERANCE;
```

Comment:
do we still need this is line 504 already sets a default for all entries in `posTolerance`?

Reply:
This allows a client to reset back to the default position joint tolerance